app/modules/net_info.c

```diff
@@ -0,0 +1,176 @@
+/* 
+ * http://blog.mclemon.io/esp8266-contributing-to-the-nodemcu-ecosystem
+ * https://github.com/smcl/nodemcu-firmware/blob/cc04aaf92c1c076c30ef0b0eee43b3f924137440/app/modules/test.c
+ *
+ * extracted ping function
+ * adoptet jan 2017 for commit to nodeMCU by Wolfgang Rosner
+ */
+
+
+#include "module.h"
+#include "lauxlib.h"
+#include "platform.h"
+
+#include "c_stdlib.h"
+
+#include "lwip/ip_addr.h"
+#include "espconn.h"
+#include "lwip/dns.h" 
+#include "lwip/app/ping.h"
+#include "lwip/raw.h"
+#include "c_stdio.h"
+
+
+
+static lua_State *gL = NULL;
+static int ping_callback_ref; 
+static int ping_host_count;
+static ip_addr_t ping_host_ip;
+
+
+void ping_received(void *arg, void *data) {
+    // this would require change of the interface 
+    // struct ping_msg *pingmsg = (struct ping_msg*)arg;
+    // struct ping_option *pingopt = pingmsg->ping_opt;
+    struct ping_option *pingopt = (struct ping_option*)arg;
+    struct ping_msg *pingmsg = pingopt->parent_msg ; 
+
+    struct ping_resp *pingresp = (struct ping_resp*)data;
+
+    char ipaddrstr[16];
+    ip_addr_t source_ip;
+    
+    source_ip.addr = pingopt->ip;
+    ipaddr_ntoa_r(&source_ip, ipaddrstr, sizeof(ipaddrstr));
+
+    if (ping_callback_ref != LUA_NOREF) {
+      lua_rawgeti(gL, LUA_REGISTRYINDEX, ping_callback_ref);
+      lua_pushinteger(gL, pingresp->bytes);
+      lua_pushstring(gL, ipaddrstr);
+      lua_pushinteger(gL, pingresp->seqno);
+      lua_pushinteger(gL, pingresp->ttl);
+      lua_pushinteger(gL, pingresp->resp_time);
+      lua_call(gL, 5, 0);
+    } else {
+      c_printf("%d bytes from %s, icmp_seq=%d ttl=%d time=%dms\n",
+	       pingresp->bytes,
+	       ipaddrstr,
+	       pingresp->seqno,
+	       pingresp->ttl,
+	       pingresp->resp_time);
+    }
+}
+
+
+
+static void ping_by_hostname(const char *name, ip_addr_t *ipaddr, void *arg) {
+    struct ping_option *ping_opt = (struct ping_option *)c_zalloc(sizeof(struct ping_option));
+
+    if (ipaddr == NULL) {
+      c_printf("SEVERE problem resolving hostname - network and DNS accessible?\n");
```

Comment:
Could you give me a hint on the reason, please?
Yes, I think generally it would be better to assemble a string an then only in the end decide what to do with it.
If only it were not that cumbersome in C .....

What about a mixed implementation?
C just giving the bare data, and lua doing the format?

But as I understood from your comment here
https://github.com/nodemcu/nodemcu-firmware/issues/1555#issuecomment-274259503
this is not so easily accomplished.